compiler/language_server/src/utils.rs

```diff
@@ -10,7 +10,7 @@ use itertools::Itertools;
 use lsp_types::{Diagnostic, DiagnosticSeverity, Position, Url};
 use std::ops::Range;
 
-pub fn error_into_diagnostic(db: &Database, module: Module, error: CompilerError) -> Diagnostic {
+pub fn error_into_diagnostic(db: &Database, module: Module, error: &CompilerError) -> Diagnostic {
```

Comment:
If this only accepts a borrowed error, `into` is no longer appropriate

```suggestion
pub fn error_to_diagnostic(db: &Database, module: Module, error: &CompilerError) -> Diagnostic {
```